libraries/AP_Compass/AP_Compass_Backend.cpp

```diff
@@ -98,13 +103,22 @@ void AP_Compass_Backend::correct_field(Vector3f &mag, uint8_t i)
     state.motor_offset.zero();
     if (_compass._per_motor.enabled() && i == 0) {
         // per-motor correction is only valid for first compass
-        _compass._per_motor.compensate(state.motor_offset);
+        _compass._per_motor.compensate(state.motor_offset, state.params.motor_comp_batt_index);
     } else if (_compass._motor_comp_type == AP_COMPASS_MOT_COMP_THROTTLE) {
-        state.motor_offset = mot * _compass._thr;
+#if APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_ArduPlane) || APM_BUILD_TYPE(APM_BUILD_Rover)
+#if APM_BUILD_TYPE(APM_BUILD_Rover)
+        AP_MotorsUGV *motors = AP::motors_ugv();
+#else
+        AP_Motors *motors = AP::motors();
+#endif
+        if (motors) {
+            state.motor_offset = mot * fabsf(motors->get_throttle());
```

Comment:
This should almost certainty be `motors->get_throttle_out()` for copter/plane. But that would be a change in behaviour. 

We would might also see some benefit from applying the expo and volt sailing with: `motors->apply_thrust_curve_and_volt_scaling(motors->get_throttle_out())`. That would change to assume the current is proportional to the throttle level sent to the ESC rather than the linearized one used in motors. That is how it works in the per-motor.  

I suspect we might even want to multiply by voltage again like we do in per-motor. 

`motors->apply_thrust_curve_and_volt_scaling(motors->get_throttle_out())*AP::battery().voltage(batt_index)`

Maybe new `COMPASS_MOTCT` option. And a option for the python magfit to calculate the compensation factors.....